lib/pg_ha_migrations/safe_statements.rb

```diff
@@ -273,15 +273,9 @@ def safe_create_partitioned_table(table, key:, type:, infer_primary_key: nil, **
       next unless options[:id]
 
       pk_columns = td.columns.each_with_object([]) do |col, arr|
-        if col.respond_to?(:primary_key)
-          next unless col.primary_key
+        next unless col.options[:primary_key]
 
-          col.primary_key = false
-        else
-          next unless col.options[:primary_key]
-
-          col.options[:primary_key] = false
-        end
+        col.options[:primary_key] = false
```

Comment:
This conditional logic is no longer needed, as the API for accessing the primary key is consistent across Rails 6.1 and 7.0